tests/jobs.py

```diff
@@ -491,3 +491,196 @@ def test_scheduler_process_check_heartrate(self):
         scheduler.process_dag(dag, queue=queue)
 
         queue.put.assert_not_called()
+
+    def test_scheduler_do_not_schedule_removed_task(self):
+        dag = DAG(
```

Comment:
The DAG creation code is very repetitive and I think it will be useful for future tests. Maybe you could refactor it into a helper function?